aleph-client/src/multisig.rs

```diff
@@ -0,0 +1,481 @@
+        description: &'static str,
+    ) -> Result<H256> {
+        Ok(try_send_xt(connection, xt, Some(description), Finalized)?
+            .expect("For `Finalized` status a block hash should be returned"))
+    }
+
+    /// Compose extrinsic for `multisig::approve_as_multi` call. Assumes that `author_idx` is correct.
+    fn construct_approve_as_multi(
+        &self,
+        connection: &Connection,
+        timepoint: Option<Timepoint>,
+        author_idx: usize,
+        call_hash: CallHash,
+    ) -> (ApproveAsMultiCall, Connection) {
+        let (author, other_signatories) = self.designate_representative_and_represented(author_idx);
+        let connection = connection.clone().set_signer(author);
+        let xt = compose_extrinsic!(
+            connection,
+            "Multisig",
+            "approve_as_multi",
+            self.threshold,
+            other_signatories,
+            timepoint,
+            call_hash,
+            MAX_WEIGHT
+        );
+        (xt, connection)
+    }
+
+    /// Compose extrinsic for `multisig::as_multi` call. Assumes that `author_idx` is correct.
+    fn construct_as_multi<CallDetails: Encode>(
+        &self,
+        connection: &Connection,
+        timepoint: Option<Timepoint>,
+        author_idx: usize,
+        call: UncheckedExtrinsicV4<CallDetails>,
+        store_call: bool,
+    ) -> (AsMultiCall, Connection) {
+        let (author, other_signatories) = self.designate_representative_and_represented(author_idx);
+        let connection = connection.clone().set_signer(author);
+        let xt = compose_extrinsic!(
+            connection,
+            "Multisig",
+            "as_multi",
+            self.threshold,
+            other_signatories,
+            timepoint,
+            call.function.encode(),
+            store_call,
+            MAX_WEIGHT
+        );
+        (xt, connection)
+    }
+
+    /// Compose extrinsic for `multisig::cancel_as_multi` call. Assumes that `author_idx` is correct.
+    fn construct_cancel_as_multi(
+        &self,
+        connection: &Connection,
+        timepoint: Timepoint,
+        author_idx: usize,
+        call_hash: CallHash,
+    ) -> (CancelAsMultiCall, Connection) {
+        let (author, other_signatories) = self.designate_representative_and_represented(author_idx);
+        let connection = connection.clone().set_signer(author);
+        let xt = compose_extrinsic!(
+            connection,
+            "Multisig",
+            "cancel_as_multi",
+            self.threshold,
+            other_signatories,
+            timepoint,
+            call_hash
+        );
+        (xt, connection)
+    }
+}
+
+/// Dispatch `call` as on behalf of the multisig party of `author` and `other_signatories`
+/// with threshold 1. `connection` is *not* assumed to be already signed by `author`.
+/// `other_signatories` *must* be sorted (according to the natural ordering on `AccountId`).
+pub fn perform_pato_multisig<CallDetails: Encode + Clone>(
```

Comment:
why pato?

Reply:
threshold 1 is weird, I can rename it if you find better name

Reply:
I would prefer a different name which would suggest what it is, not sure what is a good name for it.

Reply:
renamed